components/slot-fill/provider.js

```diff
@@ -61,6 +61,7 @@ class SlotFillProvider extends Component {
 			this.fills[ name ],
 			instance
 		);
+		this.resetFillOccurrence( name );
```

Comment:
I think this is the most important line. It causes the key to update in `slot` implementation. 💯